test.py

```diff
@@ -1,124 +1,527 @@
+        issue_num = 1
+        self.gh_provider.remove_label(label, issue_num)
+        
+        self.gh_provider.api_req.assert_called_with("DELETE", GithubApiProvider.remove_label_url % (self.owner, self.repo, issue_num, label), {})
+
+
+    def test_get_labels(self):
+        gh_provider = GithubApiProvider("jdm", "a453b3923e893f0383cd2893f...", self.owner, self.repo)
+        issue_num = 1
+        label1 = 'S-awaiting-review'
+        label2 = 'C-assigned'
+        gh_provider.api_req = Mock(return_value = {"header":"", "body":json.dumps([{'name':label1}, {'name':label2}])})
+        
+        self.assertEquals([label1, label2], gh_provider.get_labels(issue_num))
+        gh_provider.api_req.assert_called_with("GET", GithubApiProvider.get_label_url % (self.owner, self.repo, issue_num))
+
+
+    def test_get_diff(self):
+        gh_provider = GithubApiProvider("jdm", "a453b3923e893f0383cd2893f...", self.owner, self.repo)
+        diff = "fake diff"
+        diff_url = "https://github.com/servo/servo/pull/1.diff"
+        gh_provider.api_req = Mock(return_value = {"header":"", "body":diff})
+        
+        self.assertEquals(diff, gh_provider.get_diff("https://github.com/servo/servo/pull/1.diff"))
+        gh_provider.api_req.assert_called_with("GET", diff_url)
+
+
+    def test_set_assignee(self):
+        gh_provider = GithubApiProvider("jdm", "a453b3923e893f0383cd2893f...", self.owner, self.repo)
+        issue_num = 1
+        assignee = "jdm"
+        gh_provider.api_req = Mock(return_value = {"header":"", "body":assignee})
+
+        self.assertEquals(assignee, gh_provider.set_assignee(assignee, issue_num))
+        gh_provider.api_req.assert_called_with("PATCH", GithubApiProvider.issue_url % (self.owner, self.repo, issue_num), {"assignee":assignee})
+
+@patch('travisciapiprovider.urllib.urlopen')
+class TestTravisCiApiProvider(unittest.TestCase):
+    class FakeFile():
+        def __init__(self, contents):
+            self.contents = contents
+
+
+        def read(self):
+            return self.contents
+
+
+    def setUp(self):
+        self.travis = TravisCiApiProvider()
+
+
+    def test_get_build(self, urlopen_mock):
+        urlopen_mock.return_value = self.FakeFile(json.dumps({"Something":"here"}))
+        self.travis.get_build(1)
+        urlopen_mock.assert_called_with(TravisCiApiProvider.build_url.format(build_id=1))
+
+
+    def test_get_log(self, urlopen_mock):
+        job_id = 1
+        urlopen_mock.return_value = self.FakeFile(open('resources/single-line-comment.log').read())
+        self.travis.get_log({"matrix":[{'id':job_id}]})
+        urlopen_mock.assert_called_with(TravisCiApiProvider.log_url.format(job_id=job_id))
+
+
+    def test_get_pull_request_number(self, urlopen_mock):
+        pr_num = 1234
+        self.assertEquals(1234, self.travis.get_pull_request_number({"compare_url":"https://github.com/servo/servo/{}".format(pr_num)}))
+
+
+class TestErrorLogParser(unittest.TestCase):
```

Comment:
I don't think this test case provides us with anything besides code coverage improvement.